src/kernel/thread.c

```diff
@@ -325,19 +325,22 @@ static void switchSchedContext(void)
     if (NODE_STATE(ksReprogram)) {
         /* if we are reprogamming, we have acted on the new kernel time and cannot
          * rollback -> charge the current thread */
-        commitTime();
+        commitTime(switchedDomain);
     }
 
     NODE_STATE(ksCurSC) = NODE_STATE(ksCurThread)->tcbSchedContext;
 }
 #endif
 
-static void scheduleChooseNewThread(void)
+static bool_t scheduleChooseNewThread(void)
 {
+    bool_t switchedDomain = false;
     if (ksDomainTime == 0) {
```

Comment:
How come this condition isn't `isCurDomainExpired()` on MCS?

Reply:
I think it should be? In which case your previous point is even stronger, and also the original assertion was fine. If others agree, we can fix that while we're here.

Reply:
It probably should be. For functional correctness it doesn't matter, this weaker check is sufficient for execution safety, but for removing jitter you probably do want `isCurDomainExpired()` here. Let's make that a separate PR.

Reply:
Well Anna said on the issue that MIN_BUDGET didn't make sense.  So maybe it is isCurDomainExpired that should have MIN_BUDGET removed? : 
> I'm not sure MIN_BUDGET makes sense for domains. Because threads can block in the kernel, we need to make sure they have enough budget when they block in an IPC queue to be charged the time from kernel entry to the time they exit the kernel when they wake up, as well as the time to get into the IPC queue in the first place. This is why MIN_BUDGET is 2*kernel wcet.

> Domains have no need to block in an IPC queue. A domain just needs enough time to switch into the kernel. Once the domain is scheduled again, it's budget is replenished. If we don't care about domain overruns (since they are not sporadic servers, they are explicit timeslices) I think 0 makes the most sense for domain expiry, or some number that is the amount of time it takes to get into the kernel and read the time. 0 is probably easier.

Reply:
> this weaker check is sufficient for execution safety

I'm not sure what execution safety means, but I think this check at least needs ksConsumed somewhere as we are before commitTime?

Reply:
Indeed, you are right. That's the bit I was missing before.